### PR TITLE
Workaround for printf issue

### PR DESCRIPTION
**Pull request by TMRh20**
Workaround for printf issue identified in issue https://github.com/nRF24/RF24Gateway/issues/25

### CODE REVIEW

**Comment by TMRh20** [comment]
Funny stuff going on with the printfs, maybe this should all be changed to use `Serial.print(F(`

File: RF24Network.cpp [hunk 1]
```diff
@@ -163,17 +163,20 @@ uint8_t RF24Network::update(void)
...
-                                 millis(), header->toString()));
+                                 header->toString()));
 #if defined(RF24_LINUX)
         if (frame_size) {
-            IF_SERIAL_DEBUG_FRAGMENTATION_L2(printf_P(PSTR("%u: FRG Rcv frame size %i\n"), millis(), frame_size););
-            IF_SERIAL_DEBUG_FRAGMENTATION_L2(printf_P(PSTR("%u: FRG Rcv frame "), millis()); const char* charPtr = reinterpret_cast<const char*>(frame_buffer); for (uint16_t i = 0; i < frame_size; i++) { printf_P(PSTR("%02X "), charPtr[i]); }; printf_P(PSTR("\n\r")));
+            IF_SERIAL_DEBUG_FRAGMENTATION_L2(printf_P(PSTR("%u: "), millis()));
+            IF_SERIAL_DEBUG_FRAGMENTATION_L2(printf_P(PSTR("FRG Rcv frame size %i\n"), frame_size););
+            IF_SERIAL_DEBUG_FRAGMENTATION_L2(printf_P(PSTR("%u: "), millis()));
+            IF_SERIAL_DEBUG_FRAGMENTATION_L2(printf_P(PSTR("FRG Rcv frame "); const char* charPtr = reinterpret_cast<const char*>(frame_buffer); for (uint16_t i = 0; i < frame_size; i++) { printf_P(PSTR("%02X "), charPtr[i]); }; printf_P(PSTR("\n\r")));
```
Comment (2bndy5) [inline RF24Network.cpp:175]: I think you're missing a semi-colon after one of `printf_P()` statements.

This line is where everything goes wrong in the Linux CI build. But that doesn't mean it it's the only place that needs a semi-colon after `printf_P()`.

**Review by 2bndy5 — CHANGES_REQUESTED** [review]

**Comment by TMRh20** [comment]
> That would only work on Arduino though, right?

Yeah, not really a good solution. I'll have to pick this up tomorrow or sometime soon.

**Comment by 2bndy5** [comment]
I asked about the ISR because I think `millis()` (& `micros()`) is computed from a different memory partition. In [AVR core it is computed from CPU registers](https://github.com/arduino/ArduinoCore-avr/blob/eabd762a1edcf076877b7bec28b7f99099141473/cores/arduino/wiring.c#L65). It may be done similarly in other cores that use the unified ArduinoCore-API -- all I could find is [a forward declaration in that repo](https://github.com/arduino/ArduinoCore-API/blob/5b9faf6095c0321b8d129a71c936d72e72716e71/api/Common.h#L103).

**Review by 2bndy5 — APPROVED** [review]
The Arduino CI wasn't triggered because there were no changes to the Arduino examples. But, I'm guessing you tested this on the Arduino Nano already. All other CI is happy now, thanks!

**Comment by 2bndy5** [comment]
Before this gets released, we have to revert the changes in https://github.com/nRF24/RF24Network/commit/7a7a8543443eeb1352d9bb99264d0e8c76baa624 and put them in a separate branch

Similarly with changes in https://github.com/nRF24/RF24Mesh/commit/b774f03bc8b2ba86ea1cc65e8581a7f1ad15e8b7, they should go into a separate branch and removed from master.

Now that nrf_to_nrf is available in the Arduino Lib manager, we could also add that lib as a dependency for these network layers.

**Comment by TMRh20** [comment]
After some thought, I think we have a bigger issue here with printf not working. Printing the millis() separately kind of works, but I am guessing its just overflowing data somewhere else because I am getting strange behaviour with debug enabled now.

**Pull request merged**